gcsfs/core.py

```diff
@@ -270,6 +271,10 @@ def __init__(
             self._session = None
         self.connect(method=token)
 
+    @classmethod
+    def _strip_protocol(cls, path):
+        return infer_storage_options(path)["path"]
```

Comment:
Seems a little heavy handed. Why not:
```
for protocol in self.protocols:
    if path.startswith(protocol + "://"):
        path = path[len(protocol) + 3:]
```

Reply:
Right, and self.protocols contains both "gs" and "gcs". This is the sort of logic in the superclass, and what almost all of the other implementations do. The only difference you are after, is the remove the `rstrip("/")`, right?

Also, you'll notice that `AbstractFileSystem._strip_protocol` handles lists of paths and non-string paths, which we would lose here; that doesn't appear to break anything, but consistency is good. I wonder, would it be easier to add an option to that method, whether to remove the trailing "/"?

Reply:
Oh I see how the base class method does it. You're right, we just need to remote the `rstrip("/")` call for gcsfs.

Good point on the non-string path handling. I can also convert the path object to string first before sending it to `infer_storage_options` method. I think both the options are pretty doable at this point.

I'm still leaning towards the approach in my PR instead of adding an option to the base method, but that's mostly because adding an option to the method would take longer. That's because I'll have to make changes to fsspec, wait for the new version to release, and then make a change in gcsfs. But I can be easily convinced otherwise (you can even veto it since you are a maintainer for both repos).